test-haskell/cabal/cabal-3.2.1.0.ebuild

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CABAL_FEATURES="test-suite"
+CABAL_FEATURES+=" nocabaldep" # in case installed Cabal is broken
+inherit haskell-cabal
+
+MY_PN="Cabal"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A the test executable for dev-haskell/cabal"
+HOMEPAGE="https://www.haskell.org/cabal/"
+SRC_URI="https://hackage.haskell.org/package/${MY_P}/${MY_P}.tar.gz"
+
+PATCHES="${FILESDIR}/${PN}-3.2.1.0-disable-hackage-test.patch"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND=">=dev-haskell/fail-4.9:=[profile?] <dev-haskell/fail-4.10:=[profile?]
+	>=dev-haskell/mtl-2.1:=[profile?] <dev-haskell/mtl-2.3:=[profile?]
+	>=dev-haskell/parsec-3.1.13.0:=[profile?] <dev-haskell/parsec-3.2:=[profile?]
+	>=dev-haskell/semigroups-0.18.3:=[profile?] <dev-haskell/semigroups-0.20:=[profile?]
+	>=dev-haskell/text-1.2.3.0:=[profile?] <dev-haskell/text-1.3:=[profile?]
+	>=dev-lang/ghc-7.8.2:=
+"
+DEPEND="${RDEPEND}
+	=dev-haskell/cabal-3.2.1.0
+	test? ( >=dev-haskell/base-compat-0.11.0 <dev-haskell/base-compat-0.12
+		>=dev-haskell/base-orphans-0.6 <dev-haskell/base-orphans-0.9
+		>=dev-haskell/diff-0.4 <dev-haskell/diff-0.5
+		>=dev-haskell/integer-logarithms-1.0.2 <dev-haskell/integer-logarithms-1.1
+		>=dev-haskell/optparse-applicative-0.13.2.0 <dev-haskell/optparse-applicative-0.16
+		>=dev-haskell/quickcheck-2.13 <dev-haskell/quickcheck-2.14
+		>=dev-haskell/stm-2.4.5.0 <dev-haskell/stm-2.6
+		dev-haskell/tagged
+		>=dev-haskell/tar-0.5.0.3 <dev-haskell/tar-0.6
+		>=dev-haskell/tasty-1.2.3 <dev-haskell/tasty-1.4
+		>=dev-haskell/tasty-golden-2.3.1.1 <dev-haskell/tasty-golden-2.4
+		dev-haskell/tasty-hunit
+		dev-haskell/tasty-quickcheck
+		dev-haskell/temporary
+		>=dev-haskell/tree-diff-0.1 <dev-haskell/tree-diff-0.2 )
+"
+
+S="${WORKDIR}/${MY_P}"
+CABAL_CORE_LIB_GHC_PV="PM:8.10.1 PM:8.10.2"
```

Comment:
Should this `CABAL_CORE_LIB_GHC_PV` line be here? This indicates that the package is bundled with these GHC releases. But `cabal-3.2.1.0` [will likely be bundled with `ghc-8.10.3`](https://gitlab.haskell.org/ghc/ghc/-/issues/19070), whereas `cabal-3.2.0.0` is bundled with `ghc-8.10.{1,2}`.

Beyond that, my opinion is that I'm not personally convinced on the utility of having a new category `test-haskell` with an ebuild virtually a carbon-copy of `dev-haskell/cabal` in order to run the test-suite for `cabal` - which would require manually emerging this ebuild rather than pulling it in with `FEATURES=test` or `USE=test` automatically (although you may be able to set that up using a `PDEPEND` or something like that, to avoid circular dependencies). But to be blunt, I don't see the gain from the extra maintenance work. :)

Reply:
hrm, I eithe copied the `CABAL_CORE_LIB_GHC_PV` line from the previous version, or left whatever hackport put in there. Are you saying I should take it out altogether until we know for sure which ghc this will come bundled with?

Regarding `test-haskell`, I understand the hesitance - I guess it comes down to a choice between:

    1. Take on the extra maintenance in order to enable the test suite
    2. Move forward without the test suite
 
Honestly, `dev-haskell/cabal` is probably ubiquitous enough that if an error popped up that was missed (due to a missing test suite) it'd be caught quickly enough, so maybe it makes sense to just go with option `2`.

But will this always be the case? Or are there times where it makes sense to take on the extra maintenance burden in order to avoid other headaches?

Reply:
`hackport` won't put `CABAL_CORE_LIB_GHC_PV` in an ebuild, this has been copied over from `cabal-3.2.0.0`. Best to take that line out.

Reply:
Took the `test-haskell/cabal` thing out and left the test as `RESTRICT`